Infamous Software Vulnerabilities Article + New tags (#863)
chainguard-dev/edu: content/software-security/cves/infamous-cves.md

lead: "An overview a few of the most critical, widespread, and impactful known software vulnerabilities"
type: "article"
date: 2023-07-21T19:16:39+00:00
lastmod: 2023-07-25T17:17:48+00:00
contributors: ["Michelle McAveety"]
draft: false
tags: ["CVE", "Overview"]
images: []
menu:
  docs:
    parent: "cves"
weight: 003
toc: true
---

[Software vulnerabilities](/software-security/cves/cve-intro/) vary in their severity – some are difficult to exploit and have minimal implications, while others can be exploited easily and give an attacker significant leverage over a computer system. In cases where widely-implemented software contains high-severity vulnerabilities, the damage caused by their exploitation can affect millions of developers and services worldwide.

In this article, you will learn how the KEV Catalog tracks known exploited software vulnerabilities, and how it serves as a tool for developers and federal agencies. In addition, you will explore Log4Shell, Heartbleed, and Shellshock, three infamous software vulnerabilities which have had major impacts on software security worldwide.

## CISA KEV Catalog

> The U.S. Cybersecurity and Infrastructure Security Agency (CISA) operates the Known Exploited Vulnerabilities (KEV) Catalog, which is populated with CVEs that have existing exploits “in the wild”. The KEV Catalog serves as a tool for developers as it identifies CVEs that need to be prioritized for remediation because of their exploitability status. Federal civilian executive branch agencies [must remediate vulnerabilities](https://www.cisa.gov/news-events/directives/binding-operational-directive-22-01) present in the KEV Catalog by a due date specified by the CISA. Focusing on patching out these vulnerabilities limits the ability of attackers to find a potential known route into a system.

Reviewer: Maybe we can add agencies and things like the KEV Catalog to the Chainguard Glossary?

### Heartbleed (CVE-2014-0160)

[Heartbleed](https://nvd.nist.gov/vuln/detail/CVE-2014-0160) is a buffer over-read vulnerability in OpenSSL, a popular cryptographic library commonly used for encrypting SSL/TLS communications on the internet. The vulnerability allows an attacker to read the memory of a system without detection. As a result, cryptographic keys, credentials, and other content can be silently extracted from a server’s memory.

Heartbleed affected OpenSSL versions 1.0.1-1.0.1f (inclusive) and was discovered in 2014, about two years after the vulnerability was first introduced to OpenSSL. Due to its undetectable nature, determining if it was exploited against a particular server is difficult. It was estimated at the time of discovery that around [half a million websites](https://www.netcraft.com/blog/half-a-million-widely-trusted-websites-vulnerable-to-heartbleed-bug/) may have been vulnerable to the bug.

To learn more about the Heartbleed vulnerability, check out the [Heartbleed website](https://heartbleed.com/).

### Shellshock (CVE-2014-6271)

> [Shellshock](https://nvd.nist.gov/vuln/detail/CVE-2014-6271) is an arbitrary code execution vulnerability which went unnoticed for 25 years, existing in Bash since version 1.0.3 1989 and first being reported in version 4.3 in 2014. The vulnerability allows for commands that should be inaccessible to be executed through Bash’s function export feature. Following the initial CVE report, further CVEs were soon filed addressing additional related vulnerabilities.

Reviewer: I think there should be an "in" before 1989, based on the "in 2014" (and not part of the version number?). The second sentence reads a little clunky, I offer a suggestion below, but feel free to further tweak. 

```suggestion
[Shellshock](https://nvd.nist.gov/vuln/detail/CVE-2014-6271) is an arbitrary code execution vulnerability which went unnoticed for 25 years, existing in Bash since version 1.0.3 in 1989 and first being reported in version 4.3 in 2014. Through this vulnerability, commands that should be inaccessible can instead be executed through Bash’s function export feature. Following the initial CVE report, further CVEs were soon filed addressing additional related vulnerabilities.
```

Reviewer 2: There was supposed to be an "in" there, thank you for catching that!
